src/AutoMerge/dependency_confusion.jl

```diff
@@ -0,0 +1,50 @@
+# TODO: Add a more thorough explanation of the dependency confusion
+# vulnerability and how this guideline mitigates it.
+
+const guideline_dependency_confusion =
+    Guideline("No UUID conflict with other registries.",
+              data -> has_no_dependency_confusion(data.pkg,
+                                                  data.registry_head,
+                                                  data.public_registries))
+
+# TODO: Needs a strategy to handle connection failures for the public
+# registries. Preferably they should also be cloned only once and then
+# just updated to mitigate the effect of them being temporarily
+# offline. This could be implemented with the help of the Scratch
+# package, but requires Julia >= 1.5.
+function has_no_dependency_confusion(pkg, registry_head, public_registries)
+    # We know the name of this package but not its uuid. Look it up in
+    # the registry that includes the current PR.
+    packages = Pkg.TOML.parsefile(joinpath(registry_head, "Registry.toml"))["packages"]
```

Comment:
Not sure it really matters, but we already depend on TOML.jl anyway:
```suggestion
    packages = TOML.parsefile(joinpath(registry_head, "Registry.toml"))["packages"]
```

Reply:
I just followed some existing code but would never mind shortening the code. :-)